AppCenter/AppCenter/Internals/Context/Device/MSDeviceTracker.m

```diff
@@ -370,16 +374,33 @@ - (NSNumber *)timeZoneOffset:(NSTimeZone *)timeZone {
 }
 
 - (NSString *)screenSize {
-
 #if TARGET_OS_OSX
-  NSScreen *focusScreen = [NSScreen mainScreen];
-  CGFloat scale = focusScreen.backingScaleFactor;
-  CGSize screenSize = [focusScreen frame].size;
+
+  // Report screen resolution as shown in display settings ('Looks like' field in scaling tab).
+  NSSize screenSize = [NSScreen mainScreen].frame.size;
+  return [NSString stringWithFormat:@"%dx%d", (int)screenSize.width, (int)screenSize.height];
+#elif TARGET_OS_MACCATALYST
+
+  // macOS API is not directly avaliable on Mac Catalyst.
+  NSObject *screen = [NSClassFromString(@"NSScreen") valueForKey:@"mainScreen"];
+  if (screen == nil) {
+    CGSize screenSize = [UIScreen mainScreen].nativeBounds.size;
```

Comment:
When does this case happen on Catalyst? Is it just a failsafe?

Reply:
Although I don't have the whole context about it, I guess, yes, it is a fallback in case the reflection is not working.
Correct me if I'm wrong @MatkovIvan @Jamminroot

Reply:
Yes, just a fallback